hi-ml-multimodal/test_multimodal/image/model/test_model.py

```diff
@@ -156,3 +156,92 @@ def test_hubconf() -> None:
         if value_hub is None and value_himl is None:  # for example, class_logits
             continue
         assert torch.allclose(value_hub, value_himl)
+
+
+def test_load_biovil_weights() -> None:
+    """Test that the ``biovil`` option loads weights correctly."""
+    expected_model = get_biovil_resnet()
+
+    joint_feature_size = 128
+    biovil_model = ImageModel(img_model_type="resnet50",
+                              joint_feature_size=joint_feature_size,
+                              weights="biovil")
+
+    biovil_params = biovil_model.encoder.state_dict()
+    expected_params = expected_model.encoder.state_dict()
+
+    for (k, v), (exp_k, exp_v) in zip(
+        biovil_params.items(), expected_params.items()
+    ):
```

Comment:
Can you use `biovil_model.named_parameters()` instead?

Reply:
[Same for tests below]

Reply:
Ok, will do